### PR TITLE
Utilities: Add shred utility

### PR DESCRIPTION
**Pull request by romch007**
Initial implementation of shred. Missing things:
* `--force`, change permissions to allow writing if necessary,
* `--size=N`, shred this many bytes,
* `--zero`, add a final overwrite with zeros to hide shredding.

### CODE REVIEW

File: Userland/Utilities/shred.cpp [hunk 1]
```diff
@@ -0,0 +1,82 @@
...
+                    return 1;
+                }
+
+                if (nwritten == 0)
+                    break;
+
+                total_written += nwritten;
+            }
+
+            TRY(file->seek(0, SeekMode::SetPosition));
```
Comment (gmta) [inline Userland/Utilities/shred.cpp:72]: Shouldn't we also perform a sync after each iteration so write caching (if any) doesn't prevent data from being written to the storage?

File: Userland/Libraries/LibCore/System.cpp [hunk 2]
```diff
@@ -598,6 +598,13 @@ ErrorOr<void> ftruncate(int fd, off_t length)
     return {};
 }
 
+ErrorOr<void> fsync(int fd)
```
Comment (FireFox317) [inline Userland/Libraries/LibCore/System.cpp:601]: This should be a separate commit.

**Comment by supercomputer7** [comment]
Can we please add a manual page for this utility? Without it, the user needs to run the actual program with the help flag, which is not a good idea and is discouraged (as all new utilities are required to be documented).

**Comment by supercomputer7** [comment]
It would be sad to see this going closed. We can just add manual page ourselves though.

**Comment by AtkinsSJ** [comment]
Looks like missing manpages is the only issue, so let's merge it!
